yt_dlp/utils/_utils.py

```diff
@@ -5638,6 +5116,39 @@ def truncate_string(s, left, right=0):
     return f'{s[:left-3]}...{s[-right:] if right else ""}'
 
 
+class CaseInsensitiveDict(collections.UserDict):
```

Comment:
todo @Grub4K: backport?

Reply:
ah yep will add that. was there any other methods we needed to add from when you looked into it?